### PR TITLE
Update/migrate stats counts to tanstack

### PR DESCRIPTION
**Pull request by CodeyGuyDylan**
## Proposed changes:

* Get Stats counts using the tanstack query instead of redux
* Add a way to suppress global notice errors for certain queries

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

P2: p1HpG7-rcF-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
2. Make sure you site is **not** connected
3. Go to My Jetpack and make sure the global notice does not show an error
image
4. Connect your site and make sure the stats load correctly
image

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/components/product-cards-section/videopress-card.jsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ const useVideoPressStats = () => {
 		data: stats,
 		isLoading,
 		isError,
-	} = useSimpleQuery( 'videopress stats', { path: REST_API_VIDEOPRESS_FEATURED_STATS } );
```
Comment (robertsreberski) [inline projects/packages/my-jetpack/_inc/components/product-cards-section/videopress-card.jsx]: I think we use query name in the error notice, so I would keep it human readable.

Comment (robertsreberski) [reply projects/packages/my-jetpack/_inc/components/product-cards-section/videopress-card.jsx]: I see we suppress them later in the code, but I think it's confusing if we use it one way and the other.

File: projects/packages/my-jetpack/_inc/data/types.ts [hunk 2]
```diff
@@ -28,3 +28,11 @@ export type ProductCamelCase = ToCamelCase< ProductSnakeCase > & {
 		discountPricePerMonth: number;
 	};
 };
+
+export type WP_Error = {
+	code: string;
```
Comment (robertsreberski) [inline projects/packages/my-jetpack/_inc/data/types.ts]: If we are not sure / we don't care about other codes, we can at least declare here in the type ones we are using in the code so we will get a type help when developing.

```suggestion
	code: 'not_connected' | string;
```

**Review by robertsreberski — DISMISSED** [review]
LGTM, again some small not breaking suggestions.

File: projects/packages/my-jetpack/_inc/data/types.ts (hunk 2 above)
Comment (CodeyGuyDylan) [reply projects/packages/my-jetpack/_inc/data/types.ts]: Good idea, thanks!

File: projects/packages/my-jetpack/_inc/components/product-cards-section/videopress-card.jsx (hunk 1 above)
Comment (CodeyGuyDylan) [reply projects/packages/my-jetpack/_inc/components/product-cards-section/videopress-card.jsx]: I think I am going to move the error handling improvement to its own PR, I'll make these readable for now. I think later I am going to move these to constants so we aren't duplicating strings

**Review by robertsreberski — DISMISSED** [review]
This PR showcases how good our refactor is 😄 LGTM

**Review by robertsreberski — DISMISSED** [review]

**Review by robertsreberski — APPROVED** [review]

**Pull request merged**